rfcs/text/0000-project-completion-stage.md

```diff
@@ -0,0 +1,70 @@
+### DCP PR:
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# Project completion stage RFC
+
+## Summary
+
+This RFC proposes a process of releasing a contribution, or project, only after it has been completed by the DCP in order to maximize the value to the data consumer through reducing user confusion and minimizing errors in the final product.
```

Comment:
This is fundamentally different from how the Data Store and Data Browser currently operate (i.e. processing bundle by bundle).  Does the DCP support a “live” vs “hybrid” vs “curated data” presentation of data? This seems like a fundamental question that must be answered first.

Reply:
I would love clear definition of what people think live/curated/hybrid models are, and an explanation of how this RFC would impact any of that.
The DCP is not currently doing any curation, nor is any proposed in this RFC.

Reply:
How about "batch" or "embargoed" model instead of "curated"? Looking for a one word description of what this RFC is proposing. - i.e. "display only completed data".

Reply:
"embargo" has a very specific meaning and batch sounds rather techie, like the punched cards @theathorn use to carry around.

From a user perspective,  it is about the completion of an experiment.  The fact that we chop it up into a batch of bundles is something they shouldn't care about.  An experiment comes in and the results should come out as a whole not tweeted out in bundles.

I was using the term "complete submission"  however that caused confusion.  I was continually reminded that nothing is complete, as it can be updated, so I started using "version-complete", which was clear.   Then submission caused confusion since submission means a very specific thing in ingest and is not how consumers really think about obtaining data.  I believe people want the results of an experiment, although we don't yet have an experiment concept in the DCP, there is a discussion about adding it.

I see the concept as "version-complete experiment",  but probably needs simpler name, however, I am at a loss for that name.

Reply:
Many people associate "embargoed" with databases that hold data private for months while the contributor publishes.
I'm also going to get nit-picky on "live". We're not even sniffing the data until it's months or years old.
No matter what you call it, I'm not understanding how this changes the DCP model. Currently, the wranglers and the submitters go back-and-forth for weeks correcting, refining, validating submissions - all completely hidden from the public view. This RFC is just proposing that we extend that period to include the data lifecycle within the whole of the DCP, allowing us to correct, refine, validate _all_ DCP products.

Reply:
Currently, everything goes "DCP-live' bundle by bundle, sans maybe timing issues on Azul updates.

So we can have 1/2 of the primary data in the browser and part of the with analysis.

I don't really know what happens with stating before submitting to the DSS

So I don't think this is changing anything fundamental, more than formalizing it.  We currently don't even have the terminology to talk about such things.

Reply:
I like the "version-complete" concept. By ingest design, submissions **are** complete once they are submitted, though. To add more data to a project requires a second submission. See my suggestion, below, for more on this.

And, I would do more than nit-pick on "live". The idea that anything is "live", simply because we have locked together and automated the processing and release of bundles, is a fallacy. We expend considerable effort chasing this level of automation when in fact data and metadata is bottlenecked for months before even first contact with the DCP. Even _after_ first contact, it is still weeks whilst researchers in labs prepare spreadsheets and data uploads (as @jahilton nicely describes). Once prepared, data arrives in coarse-grained submissions. Arbitrarily breaking apart these coarse-grained but useful units and streaming the pieces without providing any signal of how many to expect is worse than useless.

Reply:
I think this is the key point: 

> So I don't think this is changing anything fundamental, more than formalizing it

I think it's worth recording the historic context: When the DCP was conceived, it **was** thought that users would bring data directly from their sequencers to the DCP, and many decisions were made with aims to support such a workflow, including, for exapmle, separating SS2 experiments into independent bundles and supporting "live" workflows. As Tony notes above, we've since observed that this assumption was wrong, and this RFC formalizes those observations. In so doing, it balances the advantages of rapid processing against data quality assurance.

Reply:
> The idea that anything is "live", simply because we have locked together and automated the processing and release of bundles, is a fallacy. We expend considerable effort chasing this level of automation...

I would go beyond "considerable" effort. Our entire technical architecture, and the resulting complexity, has been shaped by the requirement to stream data out as quickly as possible.

Reply:
I would not use "embargo" for these functionalities, this is still a live system just controlled in the release and understandable in its state. We may need embargo at some point but we do not have use cases here to motivate that level of delay.

Reply:
This is great discussion and it seems to have reached a plateau. I will resolve this thread. Please renew comments if there are outstanding issues that I can address in the RFC